remove the cheat flag from marine backpacks

mithrand0 opened the pull request:
The admin(s) of the following (ranked) servers:

- [CN] [GuangZhou] GAMER Rank Server
- [CN] [ShangHai] GAMER Rank Server
- [CN] GAMER 100tick Home Server
- [CN] Moon and Stars Server  Official Rank Server

have expressed interest in enabling marine backpacks by default. Currently, the `rd_server_marine_backpacks` convar is flagged `fcvar_cheat`, which prevents modification. I advised the server owners against using tools like `sm_qcvar` to bypass this restriction.

Since enabling marine backpacks does not impact gameplay in any way, nor does it give competitive advantage, I think there is no reason to flag it as cheat. 

Removing `fcvar_cheat` from it, would allow server admins to toggle the feature.

mithrand0 committed:
remove the cheat flag from marine backpacks

geotavros commented:
As I understand there are players who:
- Like backpack
- Dislike it, or their marine skins look weird with backpack (e.g. Pony Marines etc. have a backpack weapon hovering in air behind marine)
- Don't care

I think the final decision whether to draw backpack or not should be decided by Client (e.g. the player), the server can, however, suggest the default value.

So I'd suggest rd_client_marine_backpacks to have 3 states:
- -1, the default, means that backpack visibility will be controlled by rd_server_marine_backpacks
- 0 disables backpacks even if server enables rd_server_marine_backpacks 1
- 1 enables backpacks even if server disables them through rd_server_marine_backpacks 0

GmaOrange commented:
Im a backpack lover and have some friends who like it even with weird skins so im fine with change. However dmitriy's suggestion look logical.

mithrand0 committed:
implement the following logic for rd_client_marine_backpacks:

-1, the default, means that backpack visibility will be controlled by rd_server_marine_backpacks

0 disables backpacks even if server enables rd_server_marine_backpacks 1

1 enables backpacks even if server disables them through rd_server_marine_backpacks 0

mithrand0 committed:
fix description

mithrand0 commented:
> So I'd suggest rd_client_marine_backpacks to have 3 states:
> 
> * -1, the default, means that backpack visibility will be controlled by rd_server_marine_backpacks
> * 0 disables backpacks even if server enables rd_server_marine_backpacks 1
> * 1 enables backpacks even if server disables them through rd_server_marine_backpacks 0

Implemented @geotavros suggestion.
